src/pages/en/core-concepts/routing.md

```diff
@@ -34,119 +34,81 @@ Astro uses standard HTML [`<a>` elements](https://developer.mozilla.org/en-US/do
-
-Routes can be generated from multiple named parameters, at any level of the filepath:
+This will generate three pages: `dogs/clifford`, `dogs/rover`, and `dogs/spot`, each displaying the corresponding dog name.
 
-- `pages/blog/[slug].astro` → (`/blog/hello-world`, `/blog/post-2`, etc.)
-- `pages/[username]/settings.astro` → (`/fred/settings`, `/drew/settings`, etc.)
-- `pages/[lang]-[version]/info.astro` → (`/en-v1/info`, `/fr-v2/info`, etc.)
+The filename can include multiple parameters, which must all be included in the `params` objects in `getStaticPaths()`:
 
-#### The `Astro.params` object
-
-Astro components that generate routes dynamically have access to an `Astro.params` object for each route. This allows you to use those generated parts of the URL in your component script and template.
-
-```astro / (id) |{id}/ /(?<!//.*)Astro.params/
+```astro title="src/pages/[lang]-[version]/info.astro"
 ---
-// Example: src/pages/posts/[id].astro
-const { id } = Astro.params;
----
-<p>Post: {id}</p>
-
-
-// Astro.params object passed for the route `/post/abc`
-{ "id": "abc" }
-```
-
-Multiple dynamic route segments can be combined to work the same way.
+export function getStaticPaths () {
+ return [
+    {params: {lang: 'en', version: 'v1'}},
+    {params: {lang: 'fr', version: 'v2'}},
+  ];
+}
 
-```astro /(?<=const.*)(id|comment)/
+const { lang, version } = Astro.params;
 ---
-// Example: src/pages/post/[id]/[comment].astro
-const { id, comment } = Astro.params;
----
-
-// Astro.params object passed for the route `/post/abc/a-comment`
-{ "id": "abc", "comment": "a-comment" }
+...
 ```
 
-### Rest parameters
+This will generate `/en-v1/info` and `/fr-v2/info`.
 
-If you need more flexibility in your URL routing, you can use a rest parameter in your `.astro` filename as a universal catch-all for file paths of any depth by adding three dots (`...`) inside your brackets.
+Parameters can also be part of enclosing folder names, so we could use `src/pages/[lang]/[version]/info.astro` with the same `getStaticPaths` to generate `/en/v1/info` and `/fr/v2/info`.
 
-For example:
-
-- `pages/post/[...slug].astro` → (`/post/a`, `/post/a/b`, `/post/a/b/c`, etc.)
-
-Matched parameters will be passed as a query parameter (`slug` in the example) to the page.
-
-```json
-// Astro.params object passed for the route `/post/a/b/c`
-{ "slug": "a/b/c" }
-```
+📚 Learn more about [`getStaticPaths()`](/en/reference/api-reference/#getstaticpaths).
 
-:::tip
-Rest parameters are optional by default, so `pages/post/[...slug].astro` could match `/post/` as well.
-:::
+#### Rest parameters
 
-#### Example: Rest parameters
+If you need more flexibility in your URL routing, you can use a rest parameter (`[...param]`) in your `.astro` filename to match file paths of any depth:
 
-For a real-world example, you can implement GitHub’s file viewer with the following named and rest paramenters:
+```astro title="src/pages/sequences/[...path]"
```

Comment:
We need an extension here right?
````suggestion
```astro title="src/pages/sequences/[...path].astro"
````